ocp_utilities/infra.py

```diff
@@ -354,3 +357,32 @@ def create_icsp(icsp_name, repository_digest_mirrors):
     )
     icsp.deploy()
     return icsp
+
+
+def update_pull_secret(new_content_dict, secret_name, secret_namespace):
+    secret = Secret(name=secret_name, namespace=secret_namespace)
+
+    if secret.exists:
+        secret_data_dict = json.loads(
+            base64.b64decode(secret.instance.data[".dockerconfigjson"])
+        )["auths"]
+        secret_data_dict.update(new_content_dict["auths"])
+        secret_data_dict_encoded = base64.b64encode(
+            str({"auths": secret_data_dict}).replace("'", '"').encode("ascii")
+        ).decode("utf-8")
+
+        ResourceEditor(
+            patches={secret: {"data": {".dockerconfigjson": secret_data_dict_encoded}}}
+        ).update()
+    else:
+        secret_data_encoded = base64.b64encode(
+            str(new_content_dict).replace("'", '"').encode("ascii")
```

Comment:
why `("'", '"')` is needed?

Reply:
replaced with `json.dumps` ; the reason was that python dicts may use single/double quotes for key values while json pattern (which .dockerconfigjson value supports for encoding) only accepts double quotes. using `replace` was a bad w/a